# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1203586227 18000
#      Thu Feb 21 04:30:27 2008 -0500
# Branch release-3-0-x
# Node ID 86bf8c2c3fc19d8613c8d6671a6b7b96d7017fea
# Parent  4bfe1ab20e995abe1b9b9c79570dd7119307328b
imshow.m: use axis ("image")

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,13 @@
 2008-02-21  John W. Eaton  <jwe@octave.org>
 
+	* image/imshow.m: Call axis ("image").
+	From Michael G. Ross <mgross@MIT.EDU>.
+
 	* plot/errorbar.m: If nargout > 0, return vector of handles to
 	line objects.
 	* plot/__go_draw_axes__.m: Improve handling of line style for
 	errorbar plots.
 
 2008-02-20  John W. Eaton  <jwe@octave.org>
 
 	* strings/strcat.m: Detect cellstr args.
diff --git a/scripts/image/imshow.m b/scripts/image/imshow.m
--- a/scripts/image/imshow.m
+++ b/scripts/image/imshow.m
@@ -161,16 +161,17 @@ function h = imshow (im, varargin)
   endif
 
   if (true_color)
     tmp = __img__ ([], [], im);
   else
     tmp = image (round ((rows (colormap ()) - 1) * im));
   endif
   set (gca (), "visible", "off");
+  axis ("image");
 
   if (nargout > 0)
     h = tmp;
   endif
 
 endfunction
 
 %!error imshow ()                           # no arguments
